### DISCUSSION TITLE
Storage using spaces from DigitalOcean

### DISCUSSION THREAD

**Discussion by ibonkonesa** [outline/outline#2635] [General]
Hi everyone,

i tried to set up outline using a Spaces instead of AWS s3. Despite files are uploading, Ia POST request is made after create the attachment adding to the bucket_upload_url the name of the bucket.

That's my AWS config in .env file

AWS_REGION=us-east-1
AWS_S3_UPLOAD_BUCKET_URL=https://ams3.digitaloceanspaces.com
AWS_S3_UPLOAD_BUCKET_NAME=outline-test
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=false
AWS_S3_ACL=private

Regards

**Comment by inyerade** [comment]
I use this configuration:
      
      AWS_ACCESS_KEY_ID=<KEY>
      AWS_SECRET_ACCESS_KEY=<SECRET>
      AWS_REGION=sfo3
      AWS_S3_ACCELERATE_URL=https://test.sfo3.digitaloceanspaces.com
      AWS_S3_UPLOAD_BUCKET_URL=https://test.sfo3.digitaloceanspaces.com
      AWS_S3_UPLOAD_BUCKET_NAME=test
      AWS_S3_UPLOAD_MAX_SIZE=26214400
      AWS_S3_FORCE_PATH_STYLE=false

test is the name of the bucket, you have to change according your space

and also I have to configure de CORS in the space settings 

Also important is the AWS_S3_ACCELERATE_URL. 

I do not know much about S3 and spaces but maybe if AWS_S3_ACCELERATE_URL is empty the default value could be AWS_S3_UPLOAD_BUCKET_URL. I have to debug the db, see the correct attachments.url but when debug the /api/attachments.redirect?id=XXXXXX  I see a redirection to  amazon and then try setting AWS_S3_ACCELERATE_URL and works.

**Reply by denismorozov** [reply]
Thanks for this. Adding the "accelerate url" worked for me. Uploading a profile picture worked w/o it, but was getting the same error when creating a doc.

In-case it's helpful for others I posted my working config here: https://github.com/outline/outline/discussions/5326#discussioncomment-7227410
